### PR TITLE
Pract.02: Add 04-07 | Nikita Beliy

### PR DESCRIPTION
**Pull request by N1k1ta-white**
Nikita Beliy
SI 1 9MI800022

### CODE REVIEW

File: Task04.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+struct Pair {
+    int a = 0, b = 0;
+};
+
+struct Relation {
+    int size = 0;
+    Pair pairs[25];
+};
+
+Pair createPair(const int &a = 0, const int &b = 0) {
```
Comment (khatoni) [inline Task04.cpp]: do not use & with primitive data types!

File: Task04.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    Pair res{a, b};
+    return res;
+}
+
+void initPair(Pair &pair, const int &a, const int &b) {
```
Comment (khatoni) [inline Task04.cpp]: why & with int

File: Task04.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    pair.a = a;
+    pair.b = b;
+}
+
+void writePairToFile(std::fstream &file, const Pair &pair) {
+    file << pair.a << " " << pair.b << " ";
+}
+
+Pair readPairFromFile(std::fstream &file, const int &mode = 0) {
```
Comment (khatoni) [inline Task04.cpp]: const int

File: Task04.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            time++;
+            continue;
+        }
+        number = number * 10 + (int) rotator - 48;
+    }
+    if (mode == 0) {
+        file.clear();
+        file.seekg(curr_pos);
+    }
+    return res;
```
Comment (khatoni) [inline Task04.cpp]: you can just write
int a; int b;
file>>a>>b

File: Task06.cpp [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+#include <fstream>
+#include <cassert>
+
+const char PATH[] = R"(C:\Users\Nikita\CLionProjects\untitled\practice2\Task06.cpp)";
+
+int main() {
+    std::fstream me(PATH, std::ios::in);
+    assert(me.is_open());
+    char reader;
+    while (me >> std::noskipws >> reader)
```
Comment (khatoni) [inline Task06.cpp]: std::noskipws??

File: Task07.cpp [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+void error_input() {
+    std::cout << "Incorrect input\n";
+}
+
+void writeUserToFile(std::fstream &file, const User &user) {
+    file << user.email << " " << user.password << " " << user.name << std::endl;
+}
+
+User readUserFromFile(std::fstream &file, const int &mode = 0) {
```
Comment (khatoni) [inline Task07.cpp:27]: again int &, why

File: Task07.cpp [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+bool strEqual(const char str1[], const char str2[]) {
+    int i = 0;
+    while (str1[i] != '\0' && str2 != "\0" && str1[i] && str2[i]) {
+        if (str1[i] != str2[i])
+            return false;
+        i++;
+    }
+    return true;
```
Comment (khatoni) [inline Task07.cpp:72]: if the strings are with different sizes

**Review by khatoni — CHANGES_REQUESTED** [review]

**Comment by Justsvetoslavov** [comment]
Check uploaded solutions.
You can re-open the PR or open a new one if you have better solution or questions.

**Pull request closed without merging**